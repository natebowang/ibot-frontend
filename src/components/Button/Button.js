import React, {useState} from "react";
import Radium from "radium";

/**
 * Primary button component
 * @type {function(...[*]=)}
 */
const Button = Radium(({style, onClick: clickHandler, children}) => {
    return (
        <button style={{...buttonStyle, ...style}} onClick={clickHandler}>
            {children}
        </button>
    )
});

const buttonStyle = {
    border: 'none',
    borderRadius: '2px',
    padding: '1px 5px',
    cursor: 'pointer',
    // ':hover': {
    //     filter: 'brightness(85%)',
    // },
    ':active': {
        filter: 'brightness(85%)',
    },
    ':focus': {
        border: 'none',
        outline: '0',
        boxShadow: 'none',
    }
};

export default Button;

/**
 * Button component for test
 * @returns {*}
 * @constructor
 */
export const ButtonExample = () => {
    const [counter, setCounter] = useState(0);

    const style = {
        backgroundColor: 'orange',
    };

    const clickHandler = (e) => {
        e.preventDefault();
        setCounter(counter + 1);
    };

    return <Button onClick={clickHandler} style={style}>{counter}</Button>
};
