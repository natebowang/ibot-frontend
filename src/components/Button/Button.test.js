import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from "react";
import {TestButton} from "./Button";

test("Primary button snapshot test", () => {
    const button = renderer.create(
        <TestButton/>
    );

    let tree = button.toJSON();
    expect(tree).toMatchSnapshot();
});
