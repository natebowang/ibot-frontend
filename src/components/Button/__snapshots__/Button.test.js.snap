// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary button snapshot test 1`] = `
<button
  data-radium={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "orange",
      "border": "none",
      "borderRadius": "2px",
      "cursor": "pointer",
      "padding": "1px 5px",
    }
  }
>
  0
</button>
`;
